test/chttpd_httpd_handlers_tests.erl

```diff
@@ -0,0 +1,54 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License.  You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(chttpd_httpd_handlers_tests).
+
+-export([handlers/1]).
+
+-include_lib("couch/include/couch_eunit.hrl").
+
+handlers(url_handler) ->
+    [
+        {<<"">>, chttpd_misc, handle_welcome_req},
+        {<<"favicon.ico">>, chttpd_misc, handle_favicon_req},
+        {<<"_utils">>, chttpd_misc, handle_utils_dir_req},
+        {<<"_all_dbs">>, chttpd_misc, handle_all_dbs_req},
+        {<<"_active_tasks">>, chttpd_misc, handle_task_status_req},
+        {<<"_node">>, chttpd_misc, handle_node_req},
+        {<<"_reload_query_servers">>, chttpd_misc, handle_reload_query_servers_req},
+        {<<"_replicate">>, chttpd_misc, handle_replicate_req},
+        {<<"_uuids">>, chttpd_misc, handle_uuids_req},
+        {<<"_session">>, chttpd_auth, handle_session_req},
+        {<<"_up">>, chttpd_misc, handle_up_req},
+        {<<"anything">>, chttpd_db, handle_request}
```

Comment:
It would be trivial with elixir macro :-).

On a serious note though few ideas to consider (crazy ones as well).
1. Use https://github.com/richcarl/merl to generate `handlers` clauses
2. Extract the info from beam chunks somehow
3. Use macro somehow (very problematic with erlang text based preprocessor).
4. Define handlers as it is done in tests (performance impact).
5. Require something like `endpoints(db_handler) -> [<<"_view_cleanup">>, <<"compact">>]` in `<app>_httpd_handlers.erl`